jena-arq/src/main/java/org/apache/jena/query/DatasetFactory.java

```diff
@@ -254,16 +254,48 @@ public static Dataset assemble(String filename, String resourceURI) {
 	}
 
 	/**
-	 * Assemble a dataset from the model
+     * Assemble a dataset from a specific resource in a model.
+     *
+     * @param model The Model
+     * @param resourceURI The URI for the the dataset in the assembler model.
+     * @return Dataset
+     */
+	public static Dataset assemble(Model model, String resourceURI) {
+        Objects.requireNonNull(model, "model can not be null") ;
+        Objects.requireNonNull(resourceURI, "resourceURI can not be null") ;
+        Resource r = model.createResource(resourceURI);
+        return assemble(r);
+	}
```

Comment:
The indenting appears to be wrong looking at the GitHub UI diff. Probably tabs & spaces used in the same file, so no need to change I think.

Reply:
Thanks.  GH UI is the place when tabs vs spaces isn't a realistic choice. Spaces win.